wurst/objects/abilities/Omnicure.wurst

```diff
@@ -40,19 +46,11 @@ class Omnicure extends ChannelAbilityPreset
         this.setArtDuration(1, 0)
 
 
-@compiletime function createOmnicure()
-    new Omnicure(ABILITY_OMNICURE, "E", new Pair(1, 0))
+public let OMNICURE = compiletime(new Omnicure(ABILITY_OMNICURE, "E", new Pair(1, 0), MANACOST))
 
 init
-    registerSpellEffectEvent(ABILITY_OMNICURE) ->
-        let heroes = GetUnitsInRectMatching(bj_mapInitialPlayableArea, Condition(() -> begin
-            let filterUnit = GetFilterUnit()
-            return filterUnit.isType(UNIT_TYPE_HERO) and filterUnit.getOwner().isAllyOf(GetSpellAbilityUnit().getOwner())
-        end))
-
-        let owner = GetSpellAbilityUnit().getOwner()
-        for hero in heroes
-            new DummyCaster()
-                ..owner(owner)
-                ..origin(hero.getPos() + vec2(100, 0))
-                ..castTarget(ABILITY_CURE_ALL, 1, Orders.autodispel, hero)
+    EventListener.onCast(ABILITY_OMNICURE) (unit caster) ->
+        let owner = caster.getOwner()
+        forUnitsInRect(bj_mapInitialPlayableArea) (unit u) ->
```

Comment:
This needs a `nullTimer` doesn't it? Does the omnicure potion work with this current implementation? I would think it runs into the same problem that Cure All had where the banish wasn't being cast.

Reply:
Yeah, I tested and it worked.